ci: generate datalog

priv-kweihmann opened the pull request:
to keep track of the changes in the scrapped upstream data.
This will provide a document called datalog which for every version notes the newly added, the removed and the changed items in the matrix file

Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>

kkrolczyk commented on .github/workflows/cron.yml:
```diff
@@ -50,14 +51,19 @@ jobs:
         pytest
       shell: bash
+    - name: generate datalog
+      if: steps.new-commits.outputs.NEW_COMMITS == '1'
```
so, if more than one commit in MR then it should be ignored?

priv-kweihmann replied:
This is the Cron pipeline - it won't run on PRs - and what it does is using git porcelain to check if there are local uncommitted changes - and in case there are these step should be run

kkrolczyk replied:
ah, indeed, nice

kkrolczyk commented on scripts/datalog:
```diff
@@ -0,0 +1,69 @@
+    changed_values = set()
+    new_items = set()
+    removed_items = set()
+
```
this could use re.compile

and there doesn't seem to be any tests ;)

kkrolczyk commented:
i ran a quick test:
```shell
git checkout HEAD~10 osadl_matrix/osadl-matrix.json && cp osadl_matrix/osadl-matrix.json osadl-matrix.bak
git restore --staged osadl_matrix/osadl-matrix.json
./scripts/datalog osadl_matrix.bak osadl_matrix/osadl-matrix.json 1.0.0 DATALOG.md
```

with result

```
Traceback (most recent call last):
  File "./osadl-matrix/./scripts/datalog", line 69, in <module>
    main()
  File "./osadl-matrix/./scripts/datalog", line 41, in main
    changed_values.add(f'{path[0]} -> {path[1]}: changed from "{value.t1}" to "{value.t2}"')
IndexError: list index out of range
```

priv-kweihmann commented:
> i ran a quick test:
> 
> ```shell
> git checkout HEAD~10 osadl_matrix/osadl-matrix.json && cp osadl_matrix/osadl-matrix.json osadl-matrix.bak
> git restore --staged osadl_matrix/osadl-matrix.json
> ./scripts/datalog osadl_matrix.bak osadl_matrix/osadl-matrix.json 1.0.0 DATALOG.md
> ```
> 
> with result
> 
> ```
> Traceback (most recent call last):
>   File "./osadl-matrix/./scripts/datalog", line 69, in <module>
>     main()
>   File "./osadl-matrix/./scripts/datalog", line 41, in main
>     changed_values.add(f'{path[0]} -> {path[1]}: changed from "{value.t1}" to "{value.t2}"')
> IndexError: list index out of range
> ```
> 
> seems that path can be also single: `['timestamp']`

Yup I just fixed that
